db/dbScript.sql

```diff
@@ -0,0 +1,393 @@
+  FOREIGN KEY (user_id) REFERENCES Account(id),
+  FOREIGN KEY (gym_id) REFERENCES Gym(id)
+);
+create index if not exists idx_rating_gym_id on Rating(gym_id);
+
+CREATE TABLE if not exists Congestion_rating
+(
+  created_at TIMESTAMP NOT null check (created_at <= NOW()),
+  changed_at TIMESTAMP NOT null check (changed_at > created_at and changed_at <= NOW()),
+  visit_time TIME NOT NULL, --time of the account's owner visit
+  weekday INT NOT null check (weekday between 0 and 6),
+  avg_waiting_time INT NOT null check (avg_waiting_time between 1 and 5),
+  crowdedness INT NOT null check (crowdedness between 1 and 5),
+  gym_id uuid NOT NULL,
+  user_id uuid NOT NULL,
+  PRIMARY KEY (gym_id, user_id),
+  FOREIGN KEY (gym_id) REFERENCES Gym(id),
+  FOREIGN KEY (user_id) REFERENCES Account(id)
+);
+create index if not exists idx_congestion_rating_gym_id on Congestion_rating(gym_id);
+
+CREATE TABLE if not exists Gym_working_hours
+(
+  weekday INT NOT null check (weekday between 0 and 6),
+  changed_at TIMESTAMP check (changed_at <= NOW()),
+  gym_id uuid NOT NULL,
+  working_hours_id uuid NOT NULL,
+  PRIMARY KEY (weekday, gym_id, working_hours_id),
+  FOREIGN KEY (gym_id) REFERENCES Gym(id),
+  FOREIGN KEY (working_hours_id) REFERENCES Working_hours(id)
+);
+create index if not exists idx_gym_working_hours_gym_id on Gym_working_hours(gym_id);
+
+CREATE TABLE if not exists Bookmark
+(
+  created_at TIMESTAMP NOT null check(created_at <= NOW()),
+  user_id uuid NOT NULL,
+  gym_id uuid NOT NULL,
+  PRIMARY KEY (user_id, gym_id),
+  FOREIGN KEY (user_id) REFERENCES Gym(id),
+  FOREIGN KEY (gym_id) REFERENCES Account(id)
+);
+
+CREATE TABLE if not exists Notification
+(
+  id uuid NOT null DEFAULT uuid_generate_v4(),
+  type not_type NOT NULL,--more of an expansion for the potential future elaboration. for now we need just the 'message' type
+  message TEXT NOT NULL,
+  created_at TIMESTAMP NOT null check (created_at <= NOW()),
+  read_at DATE check (read_at > created_at and read_at <= NOW()), --time at which the user read/opened the notification
+  user_id uuid NOT NULL,
+  PRIMARY KEY (id),
+  FOREIGN KEY (user_id) REFERENCES Account(id)
+);
+create index if not exists idx_notification_user_id on Notification(user_id);
+
+--Gyms can mark unavailability of the gyms for a certain time period
+CREATE table if not exists Availability
+(
+  created_at TIMESTAMP NOT null check (created_at <= NOW()),
+  start_time TIMESTAMP NOT NULL,
+  end_time timestamp check (end_time > start_time and end_time > NOW()),
+  changed_at TIMESTAMP check (changed_at > created_at and changed_at <= NOW()),
+  gym_id uuid NOT NULL,
+  marked_by uuid NOT NULL,
+  PRIMARY KEY (gym_id, marked_by),
+  FOREIGN KEY (gym_id) REFERENCES Gym(id),
+  FOREIGN KEY (marked_by) REFERENCES Account(id)
+);
+
+--long story short, since all types of accounts are stored in the same table, we want to ensure that for some foreign keys we have accounts of certain type
```

Comment:
Do we really need to use triggers here? In my experience, it's best practice to handle this type of logic at the application level. Supporting it in both the database and application could lead to unnecessary complexity.